core/chains/evm/client/chain_client.go

```diff
@@ -0,0 +1,270 @@
+package client
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	commonclient "github.com/smartcontractkit/chainlink/v2/common/chains/client"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+var _ Client = (*chainClient)(nil)
+
+// TODO-1663: rename this to client, once the client.go file is deprecated.
+type chainClient struct {
+	evmClient commonclient.Client[
```

Comment:
nit: Is genericClient a better name?
Because this type is coming rom chains/client package, not evm.